x-pack/plugins/lens/public/datatable_visualization/components/table_basic.tsx

```diff
@@ -192,6 +192,21 @@ export const DatatableComponent = (props: DatatableRenderProps) => {
     ]
   );
 
+  const alignments: Record<string, 'left' | 'right' | 'center'> = useMemo(() => {
+    const alignmentMap: Record<string, 'left' | 'right' | 'center'> = {};
+    columnConfig.columns.forEach((column) => {
+      if (column.alignment) {
+        alignmentMap[column.columnId] = column.alignment;
+      } else {
+        const isNumeric =
+          firstLocalTable.columns.find((dataColumn) => dataColumn.id === column.columnId)?.meta
+            .type === 'number';
+        alignmentMap[column.columnId] = isNumeric ? 'right' : 'left';
+      }
```

Comment:
This logic here seems to power the following behaviour:
> As long as the user didn't specify an alignment, it is right aligned for numbers, left aligned for the rest

The only issue I found is that it tries to guess the type from the "field" rather than the operation. An example of the issue can be seen with `Unique count` as column of any field:

<img width="1044" alt="Screenshot 2021-02-05 at 14 34 29" src="https://user-images.githubusercontent.com/924948/107040512-9a4bb780-67bf-11eb-97e1-8fb652a8abbe.png">

Probably it's not a blocker as `Unique count` is the only operation affected (and of course any `fullReference` operation that uses `Unique count` as sub-operation).